### PR TITLE
Implements JWT auth for REST API calls.

### PR DESCRIPTION
**Pull request by davemun**
I also changed the tests for the pre-existing X-TB-PARTNER-AUTH.

### CODE REVIEW

**Comment by davemun** [comment]
This is currently still susceptible to issue #85 and will fail the 2.6 CI tests.

File: opentok/opentok.py [hunk 1]
```diff
@@ -444,3 +446,13 @@ def get_archives(self, offset=None, count=None):
 
     def _sign_string(self, string, secret):
         return hmac.new(secret.encode('utf-8'), string.encode('utf-8'), hashlib.sha1).hexdigest()
+
+    def _create_jwt_auth_header(self, api_key, api_secret):
```
Comment (aoberoi) [inline opentok/opentok.py]: use `self.api_key` and `self.api_secret`

File: tests/test_archive.py [hunk 2]
```diff
@@ -56,7 +58,12 @@ def test_stop_archive(self):
 
         archive.stop()
 
-        expect(httpretty.last_request().headers[u('x-tb-partner-auth')]).to.equal(self.api_key+u(':')+self.api_secret)
+        claims = jwt.decode(httpretty.last_request().headers[u('x-tb-opentok-auth')], self.api_secret, algorithms=[u('HS256')])
```
Comment (aoberoi) [inline tests/test_archive.py]: validate signature?

File: tests/test_archive.py [hunk 3]
```diff
@@ -56,7 +58,12 @@ def test_stop_archive(self):
...
+        expect(claims[u('iss')]).to.equal(self.api_key)
+        expect(claims[u('ist')]).to.equal(u('project'))
+        expect(float(claims[u('exp')])).to.be.greater_than(float(time.time()))
```
Comment (aoberoi) [inline tests/test_archive.py]: is this the exact time you put in?

File: tests/test_archive.py (hunk 2 above)
Comment (aoberoi) [inline tests/test_archive.py]: can we abstract this into one method that is called for each REST test?

Comment (davemun) [inline tests/test_archive.py]: python-jose raises: JWTError – If the signature is invalid in any way.

**Comment by aoberoi** [comment]
willing to merge because the issues are related to #85, not the changes in this diff.

**Pull request merged**